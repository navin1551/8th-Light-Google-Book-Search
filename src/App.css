* {
  text-align: center;
}

body {
  margin: 0;
}

.app {
  background-color: white;
}

.book-search-heading {
  color: white;
  background-color: royalblue;
  font-size: 35px;
}
