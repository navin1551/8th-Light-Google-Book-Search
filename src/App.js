import React from "react";
import { Route } from "react-router-dom";
import HomePage from "./components/HomePage/HomePage";
import ReadingList from "./components/ReadingList/ReadingList";
import GoogleContext from "./GoogleContext";
import "./App.css";

export default class App extends React.Component {
  state = {
    books: [],
    readingList: [],
    showNoResultsText: false
  };

  updateBooks = books => {
    this.setState({
      books: books || [],
      showNoResultsText: !books
    });
  };

  addToReadingList = book => {
    let currentReadingList = this.state.readingList;
    currentReadingList.push(book);
    this.setState({
      readingList: currentReadingList
    });
  };

  specialCharError = () => {
    this.setState({
      showNoResultsText: true
    });
  };

  render() {
    const contextValue = {
      books: this.state.books,
      readingList: this.state.readingList,
      showNoResultsText: this.state.showNoResultsText,
      updateBooks: this.updateBooks,
      addToReadingList: this.addToReadingList,
      specialCharError: this.specialCharError
    };

    return (
      <GoogleContext.Provider value={contextValue}>
        <div className="app">
          <div className="book-search-heading">
            <h1 className="google-book-title">Google Book Search</h1>
          </div>
          <main>
            <Route exact path="/" component={HomePage} />
            <Route path="/reading-list" component={ReadingList} />
          </main>
        </div>
      </GoogleContext.Provider>
    );
  }
}
