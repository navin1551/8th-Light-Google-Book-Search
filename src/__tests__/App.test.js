import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import App from "../App";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
    div
  );
});

it("renders App correctly", () => {
  render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
