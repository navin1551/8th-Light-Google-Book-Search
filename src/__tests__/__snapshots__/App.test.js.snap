// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="app"
>
  <div
    className="book-search-heading"
  >
    <h1
      className="google-book-title"
    >
      Google Book Search
    </h1>
  </div>
  <main>
    <div
      className="home-page-area"
    >
      <div>
        <div
          className="search-bar-area"
        >
          <div>
            <div>
              <form
                onSubmit={[Function]}
              >
                <div
                  className="search-input-area"
                >
                  <label
                    htmlFor="Search"
                  >
                    Search:
                  </label>
                  <input
                    id="user-search"
                    name="Search"
                    onChange={[Function]}
                    placeholder=""
                    type="text"
                  />
                  <button
                    id="search-button"
                  >
                    Search
                  </button>
                  <a
                    href="/reading-list"
                    id="reading-list-link"
                    onClick={[Function]}
                  >
                    My Reading List
                  </a>
                </div>
              </form>
              <div
                className="no-value-message-area"
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          data-testid="book-list"
        >
          <p
            id="no-results-message"
          >
            
          </p>
        </div>
      </div>
    </div>
  </main>
</div>
`;
